### ISSUE TITLE
one-shot strategy model scores/ WebUI

### ISSUE THREAD

**Issue by NotSure2732** [microsoft/nni#4795]
**Describe the issue**:
Regarding the use of DARTS/ and other one-shot strategies:
Is it only possible to retrieve the best architecture, or is there a way to get a ranking of the different choices/ architectures, or even a WebUI?

**Environment**:
- NNI version:
- Training service (local|remote|pai|aml|etc):
- Client OS:
- Server OS (for remote mode only):
- Python version:
- PyTorch/TensorFlow version:
- Is conda/virtualenv/venv used?:
- Is running in Docker?:

**Configuration**:
 - Experiment config (remember to remove secrets!):
 - Search space:

**Log message**:
 - nnimanager.log:
 - dispatcher.log:
 - nnictl stdout and stderr:
 

**How to reproduce it?**:

**Comment by matluster** [comment]
I assume for most of the algorithms, the "ranking" of the architectures is implicit. For DARTS, you might use the multiplication of architecture weights, or accuracy of sub-net. For ENAS, you might use the log-probability of RL agents, or sub-net accuracy, or just directly sample a few architectures as top-k.

As this is ad-hoc and vague, I'd prefer not to expose such ability, at least not publicly. If you're interested in the details, you can always hack the strategy and look inside.

**Comment by NotSure2732** [comment]
Thanks!

> As this is ad-hoc and vague, I'd prefer not to expose such ability, at least not publicly. If you're interested in the details, you can always hack the strategy and look inside.

Does that mean the only way to get these values e.g.  for DARTS would be like this? or
How would one get the values of the multiplication of architecture weights, or the accuracy of the sub-nets?

**Comment by matluster** [comment]
This is another question highly related to algorithm.

Taking DARTS as an example, in DARTS implementation, you would see an alpha, which is a parameter dict of architecture weights. Typically, we select top-1 by taking the maximum element in each individual dimension (see export). For top-k, I think you can use some beam search or dynamic programming to get those. This is what I mean by ``multiplication of architecture weights``.

Accuracy/loss of sub-nets is another story. Fix one path and do the inference (it's like what people do in SPOS). Then calculate the loss or accuracy. To find the top-k accuracy is a bit more tricky. Maybe we can use some heuristics like evolution search (it's also what people do in SPOS).

Again, I think not many people have really done this, neither in research nor in industry field. Therefore, anything reasonable could be correct. Shared above is only my humble opinion.

**Comment by NotSure2732** [comment]
Thank you for the quick replies and insights!

**Comment by ultmaster** [comment]
I've replied in the other issue.
